src/github.com/matrix-org/go-neb/database/interface.go

```diff
@@ -0,0 +1,140 @@
+package database
+
+import (
+	"github.com/matrix-org/go-neb/api"
+	"github.com/matrix-org/go-neb/types"
+)
+
+// Storer is the interface which needs to be conformed to in order to persist Go-NEB data
+type Storer interface {
+	StoreMatrixClientConfig(config api.ClientConfig) (oldConfig api.ClientConfig, err error)
+	LoadMatrixClientConfigs() (configs []api.ClientConfig, err error)
+	LoadMatrixClientConfig(userID string) (config api.ClientConfig, err error)
+
+	UpdateNextBatch(userID, nextBatch string) (err error)
+	LoadNextBatch(userID string) (nextBatch string, err error)
+
+	LoadService(serviceID string) (service types.Service, err error)
+	DeleteService(serviceID string) (err error)
+	LoadServicesForUser(serviceUserID string) (services []types.Service, err error)
+	LoadServicesByType(serviceType string) (services []types.Service, err error)
+	StoreService(service types.Service) (oldService types.Service, err error)
+
+	LoadAuthRealm(realmID string) (realm types.AuthRealm, err error)
+	LoadAuthRealmsByType(realmType string) (realms []types.AuthRealm, err error)
+	StoreAuthRealm(realm types.AuthRealm) (old types.AuthRealm, err error)
+
+	StoreAuthSession(session types.AuthSession) (old types.AuthSession, err error)
+	LoadAuthSessionByUser(realmID, userID string) (session types.AuthSession, err error)
+	LoadAuthSessionByID(realmID, sessionID string) (session types.AuthSession, err error)
+	RemoveAuthSession(realmID, userID string) error
+
+	LoadBotOptions(userID, roomID string) (opts types.BotOptions, err error)
+	StoreBotOptions(opts types.BotOptions) (oldOpts types.BotOptions, err error)
+
+	InsertFromConfig(cfg *api.ConfigFile) error
+}
+
+// NopStorage nops every store API call. This is intended to be embedded into derived structs
+// in tests
+type NopStorage struct{}
```

Comment:
Oh, I guess you want to be able to import it from other places. Still might be nice to put it somewhere else rather than having it right next to the interface definition? YMMV